apps/api/src/app/workflows-v2/usecases/extract-variables/extract-variables.usecase.ts

```diff
@@ -68,71 +71,20 @@ export class BuildPayloadSchema {
     for (const [key, value] of Object.entries(controlValue)) {
       if (isStringTipTapNode(value)) {
         processedValue[key] = this.hydrateEmailSchemaUseCase.execute({ emailEditor: value });
+      } else if (key === 'skip' && isValidRule(value as RulesLogic<AdditionalOperation>)) {
+        const fields = extractFieldsFromRules(value as RulesLogic<AdditionalOperation>)
+          .filter((field) => field.startsWith('payload.') || field.startsWith('subscriber.data.'))
```

Comment:
why this line is needed?
I removed i locally and it works as expected.

```suggestion
```

Reply:
We don't want to create the variables for the `subscriber.foo` or `steps.foo`

Reply:
why not :) when we extract a variable from the control value we extract all of the possible variables it is a generic extractor.